From acec82d5ef2c423a7b1da16b780fc7ebf0101a2e Mon Sep 17 00:00:00 2001
From: Luciana Fujii Pontello <luciana.fujii@collabora.co.uk>
Date: Sun, 25 Sep 2011 20:18:51 -0300
Subject: [PATCH 096/119] pvrvideosink: Fix blit rectangle

Blit dest rectangle gets x and y bottom and right coordinates, instead
of width and height. The right value is calculated using the width and
height plus the left and top coordinates respectively. The coordinates
calculated before were only wrong when force-aspect-ratio was set.
---
 sys/pvr2d/gstpvrvideosink.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index 0a4cbff..368ed34 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -547,8 +547,8 @@ gst_pvrvideosink_blit (GstPVRVideoSink * pvrvideosink, GstBuffer * buffer)
 
   p_blt_3d->rcDest.left = result.x;
   p_blt_3d->rcDest.top = result.y;
-  p_blt_3d->rcDest.right = result.w;
-  p_blt_3d->rcDest.bottom = result.h;
+  p_blt_3d->rcDest.right = result.w + result.x;
+  p_blt_3d->rcDest.bottom = result.h + result.y;
 
   p_blt_3d->sSrc.pSurfMemInfo = dcontext->src_mem;
   p_blt_3d->sSrc.SurfOffset = 0;
-- 
1.7.5.4

